src/elements/content-uploader/ContentUploader.js

```diff
@@ -580,12 +580,16 @@ class ContentUploader extends Component<Props, State> {
             }
         }
 
-        this.updateViewAndCollection(updatedItems, itemUpdateCallback);
+        this.updateViewAndCollection(updatedItems, () => {
+            if (itemUpdateCallback) {
+                itemUpdateCallback();
+            }
 
-        // Automatically start upload if other files are being uploaded
-        if (view === VIEW_UPLOAD_IN_PROGRESS) {
-            this.upload();
-        }
+            // Automatically start upload if other files are being uploaded
+            if (view === VIEW_UPLOAD_IN_PROGRESS) {
```

Comment:
In these cases, where we have a callback to setState, do we need to reference the new `view` value instead?

Reply:
I think we need to, because for example, at the moment a file is added to the queue, another task failed, then we want to pause the uploading instead of continuing.